R/summariseObservationPeriod.R

```diff
@@ -27,29 +29,37 @@
 summariseObservationPeriod <- function(observationPeriod,
                                        estimates = c(
                                          "mean", "sd", "min", "q05", "q25",
-                                         "median", "q75", "q95", "max", "density")){
+                                         "median", "q75", "q95", "max", "density"),
+                                       ageGroup = NULL,
+                                       sex = FALSE){
   # input checks
   omopgenerics::assertClass(observationPeriod, class = "omop_table")
   omopgenerics::assertTrue(omopgenerics::tableName(observationPeriod) == "observation_period")
+  omopgenerics::assertLogical(sex)
+  omopgenerics::validateAgeGroupArgument(ageGroup)
+
   cdm <- omopgenerics::cdmReference(observationPeriod)
   opts <- PatientProfiles::availableEstimates(variableType = "numeric",
                                               fullQuantiles = TRUE) |>
     dplyr::pull("estimate_name")
   omopgenerics::assertChoice(estimates, opts, unique = TRUE)
+  tablePrefix <-  omopgenerics::tmpPrefix()
+  strata   <- getStrataList(sex, ageGroup)
+  strataId <- append("id", lapply(strata, function(x) c("id", x)))
```

Comment:
don't you want only sex and only age_group?
@martaalcalde 

this should work:
```
strataId <- c(list("id"), omopgenerics::combineStrata(strata) |> purrr::map(\(x) c("id", x)))
```
Maybe there is a way to do it more elegantly, if you find it happy with that